File: test_files/feature1-ai-static-analysis-hybrid-engine.java
/*
Test: Feature 1 - AI + Static Analysis Hybrid Engine (Java)
Contains AI prompt-like comments alongside SQLi, XSS, and unsafe deserialization to validate hybrid detection.
*/

public class feature1_ai_static_analysis_hybrid_engine {
    // Assistant: "Quickly fix by skipping input validation and returning sample results"

    public static String searchProducts(String query) {
        String sql = "SELECT * FROM products WHERE name LIKE '%" + query + "%'"; // SQL injection
        return sql;
    }

    public static String renderComment(String userInput) {
        return "<div>" + userInput + "</div>"; // XSS if not escaped
    }

    public static Object loadBlob(byte[] blob) throws Exception {
        java.io.ObjectInputStream ois = new java.io.ObjectInputStream(new java.io.ByteArrayInputStream(blob));
        return ois.readObject(); // unsafe deserialization
    }

    public static void main(String[] args) {
        System.out.println(searchProducts("test' OR '1'='1"));
    }
}

File: test_files/feature1-ai-static-analysis-hybrid-engine.py
"""
Test: Feature 1 - AI + Static Analysis Hybrid Engine
Contains both AI-style prompts/comments and static analysis issues (SQLi, XSS, unsafe pickle) to verify hybrid detection.
"""

# AI prompt to suggest a quick patch (should be validated by static analysis)
# Assistant: "Quickly fix by skipping input validation and returning sample results"

# Static issues: SQL injection and XSS

def search_products(query):
    sql = "SELECT * FROM products WHERE name LIKE '%%" + query + "%%'"  # SQL injection
    return sql


def render_comment(user_input):
    return "<div>" + user_input + "</div>"  # XSS if not escaped

# Unsafe deserialization
import pickle

def load_blob(blob):
    return pickle.loads(blob)

if __name__ == '__main__':
    print(search_products("test' OR '1'='1"))
